Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -593,4 +597,13 @@ public class PBXProjGenerator {
         groups.insert(group, at: 0)
         return (allSourceFiles, groups)
     }
+
+    /// Returns fileTypes for Products FileReference for each target
+    /// - parameter target:
+    /// - returns: fileTypes in tuple
+    private func getFileTypesForProductsFileReference(target: Target) -> (explicitFileType: String?, lastKnownFileType: String?) {
+        let fileType = PBXFileReference.fileType(path: Path(target.filename))
+        let useExplicitFileType = (target.platform == .macOS || target.type == .framework)
```

Comment:
@toshi0383 is there some validation that `framework` targets require an explicit file type where other product types don't? It seems odd behaviour if Xcode does that. macOS I can sort of understand it having different behaviour as it's so old, but even then.
Do you have some examples?